File: app/src/main/java/com/bitcoin/merchant/app/util/DialogUtil.java
package com.bitcoin.merchant.app.util;

import android.app.Activity;
import android.app.AlertDialog;

public class DialogUtil {
    public static void show(Activity activity, String title, String message, final Runnable runner) {
        show(activity, title, message, activity.getString(android.R.string.ok), runner);
    }

    public static void show(Activity activity, String title, String message, String positiveText, final Runnable runner) {
        activity.runOnUiThread(() -> {
            AlertDialog.Builder builder = new AlertDialog.Builder(activity);
            builder.setTitle(title)
                    .setMessage(message)
                    .setCancelable(false)
                    .setPositiveButton(positiveText, (dialog, whichButton) -> {
                        dialog.dismiss();
                        if (runner != null) {
                            runner.run();
                        }
                    });
            if (!activity.isFinishing()) {
                builder.create().show();
            }
        });
    }
}

File: app/src/main/java/com/bitcoin/merchant/app/util/DialogUtil.kt
package com.bitcoin.merchant.app.util

import android.R
import android.app.Activity
import android.app.AlertDialog
import android.content.DialogInterface

object DialogUtil {
    fun show(activity: Activity, title: String?, message: String?, runner: () -> Unit) {
        show(activity, title, message, activity.getString(R.string.ok), runner)
    }

    fun show(activity: Activity, title: String?, message: String?, positiveText: String?, runner: () -> Unit) {
        activity.runOnUiThread {
            val builder = AlertDialog.Builder(activity)
            builder.setTitle(title)
                    .setMessage(message)
                    .setCancelable(false)
                    .setPositiveButton(positiveText) { dialog: DialogInterface, _: Int ->
                        dialog.dismiss()
                        runner.invoke()
                    }
            if (!activity.isFinishing) {
                builder.create().show()
            }
        }
    }
}